build_info.json

```diff
@@ -0,0 +1,63 @@
+{
+  "comment": "Provides build info for use by downstream build rules.",
```

Comment:
Maybe just make this json5 so we can have inline comments?

Reply:
Ah yeah that's true. I imagined this being the input for an *emboss* build step that would generate GN, CMake, Bazel, Soong, etc integration files (as well as plain json), but I realize I didn't actually articulate that here.